lib/game.rb

```diff
@@ -0,0 +1,63 @@
+require_relative "./display"
+require_relative "./presenter"
+require_relative "./board"
+require_relative "./player"
+require_relative "./position_validator"
+require_relative "./parse_input"
+require_relative "./option_validator"
+require_relative "./user"
+require_relative "./tictactoe"
+
+class Game
+  attr_reader :user, :game_options, :parse_input, :validator
+
+  def initialize(user:, game_options:, parse_input:, validator:)
+    @user = user
+    @game_options = game_options
+    @parse_input = parse_input
+    @validator = validator
+  end
+
+  def players
+    game_options_descriptions = game_options.keys
+    selection = selection(game_options_descriptions)
+    game_option_description = game_options_descriptions[selection - 1]
+    game_options[game_option_description]
+  end
+
+  private
+
+  def player_prompt(game_options_descriptions)
+    options = ""
+    game_options_descriptions.each_with_index do |option, index|
+      options += "  #{index + 1}.#{option}"
+    end
+    "Please select your game choice:#{options}"
+  end
+
+  def selection(game_options_descriptions)
+    user.valid_input(
+      message: player_prompt(game_options_descriptions),
+      parse_input: parse_input,
+      validator: validator
+    )
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
+  display = Display.new($stdout, $stdin)
+  presenter = TextPresenter.new
+  board = Board.new
+  position_validator = PositionValidator.new(board)
+  parse_input = ParseInput.new
+  user = User.new(display)
+  human_player = HumanPlayer.new(user: user, token: "X", validator: position_validator, parse_input: parse_input)
```

Comment:
Here and in the `Game` we pass in the `user`, `validator` and `parse_input`. I wonder if there is a way to push all of that into a `User`? Perhaps there isn't just one `User`, but maybe there are different types? Like a user that can set up a game, and a user that can play a game? Or maybe `User` would be the right name anymore, maybe something like a `Persona` makes more sense. IDK

What that might enable us to do is to close off the way `User, `validator` and `parse_input` interact.

```ruby
class SetupGamePersona
  def initialize(validator: , parse_input:)
  end

  def valid_input(message:)
    # Usee the validator and parse_input that the object is initialized with
  end
end
```

I *think* that something like this might simplify some of the object interactions and reduce the number of direct collaborators that `Player` and `Game` require.

Reply:
Hmmm.....I'll play around with this one and see what I can come up with

Reply:
The reason I don't like SetupGamePersona is that it's naturally coupled with Game - it always goes with game.  I like User because we can plug and play, we can insert it anywhere we need it.

Reply:
I made some changes to reduce collaborators - let me know your thoughts.  If doesn't look good then I'll incorporate the personas.